packages/components/src/button/index.tsx

```diff
@@ -141,12 +141,49 @@ export function UnforwardedButton(
 
 	const trulyDisabled = disabled && ! isFocusable;
 	const Tag = href !== undefined && ! trulyDisabled ? 'a' : 'button';
+
+	const { role = 'button' } = additionalProps;
+	let ariaPropName:
+		| 'aria-pressed'
+		| 'aria-checked'
+		| 'aria-selected'
+		| undefined;
+
+	switch ( role ) {
```

Comment:
Since the `Button` component can be given an `href`, should there also be a case for `role="link"` with `aria-current`?

Reply:
Potentially, but [`aria-current`](https://w3c.github.io/aria/#aria-current) can actually be used for (almost) any item "within a container or set of related elements", so special-casing it here might not make sense. Happy to discuss though.